hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java

```diff
@@ -90,8 +90,9 @@ public OzoneDelegationTokenSecretManager(OzoneConfiguration conf,
         service, LOG);
     currentTokens = new ConcurrentHashMap();
     this.tokenRemoverScanInterval = dtRemoverScanInterval;
-    this.store = new OzoneSecretStore(conf);
-    this.s3SecretManager = s3SecretManager;
+    this.s3SecretManager = (S3SecretManagerImpl) s3SecretManager;
```

Comment:
why do we need to change s3secret manager here from interface to impl? seems not relate to this ticket.

Reply:
S3SecretManagerImpl has helper function to return OMMetadataManager required in secret store. Its interface doesn't. Since it is returning a dependency i think its better to not push it to interface itself.